includes/functions.php

```diff
@@ -2396,11 +2398,19 @@ function cache_peeringdb()
                             $peer_keep[] = dbInsert($peer_insert, 'pdb_ix_peers');
                         }
                     }
-                    $pdb_ix_peers_ids = implode(',', $peer_keep);
-                    dbDelete('pdb_ix_peers', "`pdb_ix_peers_id` NOT IN ($pdb_ix_peers_ids)");
                 }
-                $pdb_ix_ids = implode(',', $keep);
-                dbDelete('pdb_ix', "`pdb_ix_id` NOT IN ($pdb_ix_ids)");
+            }
+
+            // cleanup
+            if (empty($peer_keep)) {
+                dbDelete('pdb_ix_peers');
```

Comment:
I'm not sure you can move this out of the loop, pdb_ix_peers is grouped by ix_id so you need to delete the peers that no longer are in that IX.

Reply:
@laf we collect all ix_peers from all groups, then delete the ones we didn't see.
ix1 = 1,4,5,6
ix2 = 7,8,9,10

old code:
delete not in 1,4,5,6
delete not in 7,8,9,10

new code:
delete not in 1,4,5,6,7,8,9,10

your suggestion:
delete not in 1,4,5,6 where ix = 1
delete not in 7,8,9,10 where ix =2

Is that right?

Reply:
Good shout :)